trio/_core/_multierror.py

```diff
@@ -414,3 +418,35 @@ def trio_show_traceback(self, etype, value, tb, tb_offset=None):
                 print_exception(value)
 
             ip.set_custom_exc((BaseExceptionGroup,), trio_show_traceback)
+
+
+# Ubuntu's system Python has a sitecustomize.py file that import
+# apport_python_hook and replaces sys.excepthook.
+#
+# The custom hook captures the error for crash reporting, and then calls
+# sys.__excepthook__ to actually print the error.
+#
+# We don't mind it capturing the error for crash reporting, but we want to
+# take over printing the error. So we monkeypatch the apport_python_hook
+# module so that instead of calling sys.__excepthook__, it calls our custom
+# hook.
+#
+# More details: https://github.com/python-trio/trio/issues/1065
+if (
+    sys.version_info < (3, 11)
+    and getattr(sys.excepthook, "__name__", None) == "apport_excepthook"
+):
+    from types import ModuleType
+
+    import apport_python_hook
+    from exceptiongroup import format_exception
+
+    assert sys.excepthook is apport_python_hook.apport_excepthook
+
+    def replacement_webhook(etype, value, tb):
+        sys.stderr.write("".join(format_exception(etype, value, tb)))
```

Comment:
The original Trio hook did `for chunk in format_exception(...): sys.stderr.write(chunk)`. I sort of prefer that approach, so that if something in format_exception() raises an exception, you can at least see the traceback before the bad part. (format_exception() is a generator)

Reply:
But `format_exception()` returns a list, not a generator, so I don't see how that would help.

Reply:
Ah you're right -- the Trio hook used a generator but the exceptiongroup one doesn't. I don't think it matters enough to change it.